src/App.jsx

```diff
@@ -14,33 +15,79 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [visibleGoods, setVisibleGoods] = useState([...goodsFromServer]);
+  const [reversed, setReversed] = useState(false);
+  const [sortMethod, setSortMethod] = useState(''); // 'alphabet' | 'length' | ''
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const sortAlphabetically = () => {
+    setSortMethod('alphabet');
+    setVisibleGoods([...goodsFromServer].sort((a, b) => a.localeCompare(b)));
+    setReversed(false);
+  };
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  const sortByLength = () => {
+    setSortMethod('length');
+    setVisibleGoods([...goodsFromServer].sort((a, b) => a.length - b.length));
+    setReversed(false);
+  };
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  const reverseGoods = () => {
+    setReversed(!reversed);
+    setVisibleGoods(prevGoods => [...prevGoods].reverse());
+  };
+
+  const resetGoods = () => {
+    setSortMethod('');
+    setReversed(false);
+    setVisibleGoods([...goodsFromServer]);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={`button is-info ${sortMethod === 'alphabet' ? '' : 'is-light'}`}
```

Comment:
To set classes conditionally it's better to use [classnames](https://www.npmjs.com/package/classnames) or [clsx](https://www.npmjs.com/package/clsx) library